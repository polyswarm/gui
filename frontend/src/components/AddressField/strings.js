const strings = {
  address: 'Enter address'
};

export default strings;