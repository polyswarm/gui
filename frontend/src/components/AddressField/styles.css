.AddressField {
    display: flex;
    flex-direction: column;
    margin: 0 auto;
    width: 80%;
}

.AddressField-Label {
    color: #9c3bf8;
}

.AddressField-Input {
    background-color: #efefef;
    border: #e5e5e5 2px solid;
    border-radius: 4px;
    box-sizing: border-box;
    font-size: 24px;
    height: 36px;
    line-height: 36px;
}

.error {
    border: red 2px solid;
}

.error:focus {
    outline-color: rgba(0,0,0,0);
}