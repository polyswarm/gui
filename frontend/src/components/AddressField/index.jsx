// Vendor imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
// Project imports
// Component imports
import strings from './strings';
import './styles.css';

class AddressField extends Component {
  constructor(props) {
    super(props);
    this.state = {
      address: '',
      error: false
    }

    this.onChange = this.onChange.bind(this);
  }

  render() {
    const {state: {address, error} } = this;
    const inputClass = classNames('AddressField-Input', {error: error});
    return (
        <div className='AddressField'>
          <label className='AddressField-Label'
            htmlFor='address'>
            {strings.address}
          </label>
          <input
            className={inputClass}
            id='address'
            type='text'
            value={address}
            onChange={this.onChange}/>
        </div>
    );
  }

  onChange(e) {
    const {props: {onChange}} = this;
    const address = e.target.value;
    if (onChange) {
      onChange(address);
    }
    this.setState({address: address, error: !this.validateAddress(address)});
  }

  validateAddress(address) {
    return address.length > 14;
  }
}
AddressField.proptypes = {
  onChange: PropTypes.func,
}
export default AddressField;