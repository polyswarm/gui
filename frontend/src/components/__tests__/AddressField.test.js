import React from 'react';
import {render, shallow} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import AddressField from '../AddressField';

it('renders without crashing', () => {
  const wrapper = render(<AddressField />);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('calls onchange when instance onChange is called', () => {
  const onChange = jest.fn();
  const wrapper = shallow(<AddressField onChange={onChange}/>);
  const instance = wrapper.instance();

  instance.onChange({target:{value: 'change'}});

  expect(onChange).toHaveBeenCalledTimes(1);
  expect(onChange).toHaveBeenCalledWith('change', false);
});

it('calls onchange when address changed', () => {
  const onChange = jest.fn();
  const wrapper = shallow(<AddressField onChange={onChange}/>);
  
  wrapper.find('input').simulate('change', {target: {value: 'change'}});

  expect(onChange).toHaveBeenCalledTimes(1);
  expect(onChange).toHaveBeenCalledWith('change', false);
});

it('calls onChange with valid:true when length over 14 characters', () => {
  const onChange = jest.fn();
  const wrapper = shallow(<AddressField onChange={onChange}/>);
  
  wrapper.find('input').simulate('change', {target: {value: '012345678912345'}});

  expect(onChange).toHaveBeenCalledTimes(1);
  expect(onChange).toHaveBeenCalledWith('012345678912345', true);
});

it('calls onchange with valid:true when length is 0', () => {
  const onChange = jest.fn();
  const wrapper = shallow(<AddressField onChange={onChange}/>);
  
  wrapper.find('input').simulate('change', {target: {value: ''}});

  expect(onChange).toHaveBeenCalledTimes(1);
  expect(onChange).toHaveBeenCalledWith('', true);
});

it('sets error to true when set address is too short', () => {
  const setState = jest.spyOn(AddressField.prototype, 'setState');
  const wrapper = shallow(<AddressField />);
  setState.mockClear();
  
  wrapper.find('input').simulate('change', {target: {value: 'change'}});

  expect(setState).toHaveBeenCalledWith({error: true});
});

it('sets error to false when set address is long enough', () => {
  const setState = jest.spyOn(AddressField.prototype, 'setState');
  const wrapper = shallow(<AddressField />);
  setState.mockClear();
  
  wrapper.find('input').simulate('change', {target: {value: '012345678901234'}});

  expect(setState).toHaveBeenCalledWith({error: false});
});