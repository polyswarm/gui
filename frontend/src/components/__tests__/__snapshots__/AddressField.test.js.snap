// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="AddressField"
>
  <label
    class="AddressField-Label"
    for="address"
  >
    Enter address
  </label>
  <input
    class="AddressField-Input"
    id="address"
    type="text"
    value=""
  >
    
  </input>
</div>
`;
